qa/rpc-tests/sc_keyrot.py

```diff
@@ -0,0 +1,226 @@
+                                  amount_cert_1,
+                                  None, # no prev_cert_hash
+                                  False
+                                 )
+
+        #------------------------------------------------
+        mark_logs("## Test SC v2 (nonceasing) w/o keyrot, expect failure ##", self.nodes, DEBUG_MODE)
+        #------------------------------------------------
+        epoch_number = 0
+        quality = 0
+
+        addr_node0 = self.nodes[0].getnewaddress()
+        bwt_amount_1 = Decimal("1")
+        amount_cert_1 = {"address": addr_node0, "amount": bwt_amount_1}
+
+        self.try_send_certificate(PARAMS_NAME + "2",
+                                  False,
+                                  sc2_id,
+                                  epoch_number,
+                                  quality,
+                                  self.nodes[0].getblockcount(),
+                                  MBTR_SC_FEE,
+                                  FT_SC_FEE,
+                                  amount_cert_1,
+                                  None,
+                                  True,
+                                  "bad-sc-cert-proof"
+                                 )
+
+        #------------------------------------------------
+        mark_logs("## Test SC v2 (nonceasing) with keyrot with random key, expect failure ##", self.nodes, DEBUG_MODE)
+        #------------------------------------------------
+        self.try_send_certificate(PARAMS_NAME_KEYROT + "2",
+                                  False,
+                                  sc2_keyrot_id,
+                                  epoch_number,
+                                  quality,
+                                  self.nodes[0].getblockcount(),
+                                  MBTR_SC_FEE,
+                                  FT_SC_FEE,
+                                  amount_cert_1,
+                                  generate_random_field_element_hex(),
+                                  True,
+                                  "bad-sc-cert-proof"
+                                 )
+
+        #------------------------------------------------
+        mark_logs("## Test SC v2 (nonceasing) with keyrot, expect success ##", self.nodes, DEBUG_MODE)
+        #------------------------------------------------
+        self.try_send_certificate(PARAMS_NAME_KEYROT + "2",
+                                  False,
+                                  sc2_keyrot_id,
+                                  epoch_number,
+                                  quality,
+                                  self.nodes[0].getblockcount(),
+                                  MBTR_SC_FEE,
+                                  FT_SC_FEE,
+                                  amount_cert_1,
+                                  0, # 0 means that PHANTOM_HASH will be used
+                                  False
+                                 )
+
```

Comment:
I see here that we test passing the PHANTOM_HASH for the proof of the first certificate. Should we add more tests to cover also the generation of proof for certificates with epoch other than the first one (so that we don't pass the PHANTOM_HASH) or this scenario is already taken into account in other Python tests?

Reply:
Such scenario is indeed tested in all the other tests that send v2 sc certificates, such as `sc_cert_nonceasing.py` and `sc_getcertmaturityinfo.py`.